addons/binding/org.openhab.binding.russoundserial/src/main/java/org/openhab/binding/russoundserial/internal/RussoundSerialHandlerFactory.java

```diff
@@ -0,0 +1,89 @@
+import java.math.BigDecimal;
+import java.util.Set;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import com.thejholmes.russound.Russound;
+import com.thejholmes.russound.RussoundCommandSender;
+import com.thejholmes.russound.RussoundCommander;
+import com.thejholmes.russound.RussoundTranslator;
+import com.thejholmes.russound.Zone;
+import com.thejholmes.russound.serial.SerialCommandReceiver;
+import com.thejholmes.russound.serial.SerialCommandSender;
+import org.openhab.binding.russoundserial.handler.RussoundSerialBridgeHandler;
+import org.openhab.binding.russoundserial.handler.RussoundSerialBridgeHandler.ZoneListener;
+import org.openhab.binding.russoundserial.handler.RussoundSerialZoneHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.openhab.binding.russoundserial.RussoundSerialBindingConstants.SERIAL_BRIDGE;
+import static org.openhab.binding.russoundserial.RussoundSerialBindingConstants.ZONE_HANDLER;
+
+/**
+ * The {@link RussoundSerialHandlerFactory} is responsible for creating things and thing handlers.
+ *
+ * @author holmes - Initial contribution
+ */
+public class RussoundSerialHandlerFactory extends BaseThingHandlerFactory {
+  private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS =
+      Sets.newHashSet(SERIAL_BRIDGE, ZONE_HANDLER);
+
+  private final Logger logger = LoggerFactory.getLogger(RussoundSerialHandlerFactory.class);
+
+  @Override
+  public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+    return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+  }
+
+  @Override
+  protected ThingHandler createHandler(Thing thing) {
+    ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+    if (thingTypeUID.equals(SERIAL_BRIDGE)) {
+      String serialPort = (String) thing.getConfiguration().get("serialPort");
+      File deviceLocation = new File(serialPort);
+
+      try {
+        RussoundCommandSender commandSender = SerialCommandSender.Factory.fromFile(deviceLocation);
+        RussoundCommander commander =
+            Russound.Companion.sender("russound-serial-bridge", commandSender);
+
+        ZoneListener zoneInfoListener = new ZoneListener();
+        RussoundTranslator translator =
+            Russound.Companion.receiver("russound-serial-bridge", zoneInfoListener);
+        SerialCommandReceiver commandReceiver =
+            new SerialCommandReceiver(translator, new FileInputStream(deviceLocation));
+
+        return new RussoundSerialBridgeHandler((Bridge) thing, commander, commandReceiver,
+            zoneInfoListener);
+      } catch (FileNotFoundException e) {
+        logger.info("Couldn't find serialPort at {}", serialPort, e);
+        return null;
+      } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception